Sources/Sentry/SentryEnvelope.m

```diff
@@ -231,15 +233,23 @@ - (nullable instancetype)initWithReplayEvent:(SentryReplayEvent *)replayEvent
                                        video:(NSURL *)videoURL
 {
     NSData *replayEventData = [SentrySerialization dataWithJSONObject:[replayEvent serialize]];
-    NSData *recording = [SentrySerialization dataWithReplayRecording:replayRecording];
+    NSData *_Nullable nullableRecording =
+        [SentrySerialization dataWithReplayRecording:replayRecording];
+    if (nil == nullableRecording) {
+        SENTRY_LOG_ERROR(
+            @"Could not serialize replay recording data for envelope item. Recording will be nil.");
+        return nil;
+    }
+    NSData *_Nonnull recording = SENTRY_UNWRAP_NULLABLE(NSData, nullableRecording);
+
     NSURL *envelopeContentUrl =
         [[videoURL URLByDeletingPathExtension] URLByAppendingPathExtension:@"dat"];
 
-    BOOL success = [SentryMsgPackSerializer serializeDictionaryToMessagePack:@{
-        @"replay_event" : replayEventData,
-        @"replay_recording" : recording,
-        @"replay_video" : videoURL
-    }
+    NSMutableDictionary *envelopeContent = [NSMutableDictionary dictionary];
+    envelopeContent[@"replay_event"] = replayEventData;
+    envelopeContent[@"replay_recording"] = recording;
+    envelopeContent[@"replay_video"] = videoURL;
+    BOOL success = [SentryMsgPackSerializer serializeDictionaryToMessagePack:envelopeContent
```

Comment:
This can now be reverted:

```suggestion
    BOOL success = [SentryMsgPackSerializer serializeDictionaryToMessagePack:@{
        @"replay_event" : replayEventData,
        @"replay_recording" : recording,
        @"replay_video" : videoURL
    }
```